src/runtime/internal/utils.ts

```diff
@@ -159,5 +159,5 @@ export function set_store_value(store, ret, value = ret) {
 export const has_prop = (obj, prop) => Object.prototype.hasOwnProperty.call(obj, prop);
 
 export function action_destroyer(action_result) {
-	return action_result && is_function(action_result.destroy) ? action_result.destroy : noop;
+	return action_result ? () => is_function(action_result.destroy) && action_result.destroy() : noop;
```

Comment:
```suggestion
	return action_result && is_function(action_result.destroy) ? () => action_result.destroy() : noop;
```

Reply:
You should default to the most permissive option imo but I'm not willing to die on that hill
If it were left to me I'd just cut it to the following, there really is no reason to check anything until you really need to
```ts
return () => action_result?.destroy?.();
```